lib/stacks/lib/assets/user-data/node.sh

```diff
@@ -0,0 +1,212 @@
+
+# Update packages.
+sudo yum -y update
+sudo yum -y install time
+
+# Download cloud assets.
+echo "Downloading assets zip file"
+aws s3 cp $CLOUD_ASSETS_S3_PATH $CLOUD_ASSETS_DOWNLOAD_PATH --region $AWS_REGION
+unzip -qo $CLOUD_ASSETS_DOWNLOAD_PATH -d $CLOUD_ASSETS_PATH
+
+# TODO:
+# The "signer" and "miner" configurations will both need access to secret keys that should either be
+# produced on startup or retrieved from an existing secret and supplied to the host via an ARN. That
+# functionality should be included here.
+#
+# ```bash
+# sudo yum -y install npm
+# npm install @stacks/cli
+# sudo mkdir -p /etc/stacks
+# npx @stacks/cli make_keychain 2>/dev/null | jq > /etc/stacks/$STACKS_NODE_CONFIGURATION-keychain.json
+# ```
+
+sudo mkdir -p /etc/stacks/
+CONFIG_DIR=$CLOUD_ASSETS_PATH/stacks/config/$STACKS_NODE_CONFIGURATION
+sudo envsubst < $CONFIG_DIR/stacks.toml > /etc/stacks/stacks.toml
+
+echo "Install CloudWatch Agent"
+sudo yum -y install amazon-cloudwatch-agent
+
+echo "Configure Cloudwatch Agent"
+sudo mkdir -p /opt/aws/amazon-cloudwatch-agent/etc/
+cp $CONFIG_DIR/cw-agent.json /opt/aws/amazon-cloudwatch-agent/etc/custom-amazon-cloudwatch-agent.json
+# TODO: Publish prometheus metrics as well. We should update the dashboard template in tandem.
+
+echo "Starting CloudWatch Agent"
+amazon-cloudwatch-agent-ctl -a fetch-config -c file:/opt/aws/amazon-cloudwatch-agent/etc/custom-amazon-cloudwatch-agent.json -m ec2 -s
+systemctl status amazon-cloudwatch-agent
+
+# Set up stacks user.
+echo "Adding stacks user and group"
+sudo groupadd -g 1002 stacks
+sudo useradd -u 1002 -g 1002 -s /bin/bash stacks
+sudo usermod -aG wheel stacks
+sudo passwd -d stacks # No password.
+
+# Configure CloudFormation helper scripts. -------------------------------------
+# Note: This needs to be set up before the volumes will be picked up.
+sudo mkdir -p /etc/cfn/hooks.d/
+if [[ "$STACK_ID" != "none" ]]; then
+  echo "Configuring CloudFormation helper scripts"
+  sudo envsubst < $CLOUD_ASSETS_PATH/cfn-hup/cfn-hup.conf > /etc/cfn/cfn-hup.conf
+  sudo envsubst < $CLOUD_ASSETS_PATH/cfn-hup/cfn-auto-reloader.conf > /etc/cfn/hooks.d/cfn-auto-reloader.conf
+
+  echo "Starting CloudFormation helper scripts as a service"
+  cp $CLOUD_ASSETS_PATH/cfn-hup/cfn-hup.service  /etc/systemd/system/cfn-hup.service
+
+  systemctl daemon-reload
+  systemctl enable --now cfn-hup
+  systemctl start cfn-hup.service
+
+  cfn-signal --stack $STACK_NAME --resource $RESOURCE_ID --region $AWS_REGION
+fi
+
+# Set up volumes -----------------------------------------------------------------
+echo "Waiting for volumes to be available"
+sleep 60
+
+sudo mkdir -p $DATA_VOLUME_PATH
+
+if [[ "$DATA_VOLUME_TYPE" == "instance-store" ]]; then
+  echo "Data volume type is instance store"
+
+  sudo chmod +x $CLOUD_ASSETS_PATH/setup-instance-store-volumes.sh
+
+  (crontab -l; echo "@reboot $CLOUD_ASSETS_PATH/setup-instance-store-volumes.sh > /tmp/setup-instance-store-volumes.log 2>&1") | crontab -
```

Comment:
had you considered a systemd timer unit here vs crontab? nothing wrong with cron, just curious

Reply:
No, TBH I just used whatever I am more familiar with.